### ISSUE TITLE
Running multi queries on distributed engine at the same time but show fatal message for VM environment

### ISSUE THREAD

**Issue by tkionshao** [ClickHouse/ClickHouse#18813]
Hi Clickhouse Team,

I faced an issue in my test environment which is a virtual machine with 128GB memory. Cause our scenario is to get one aggregated result and four to five queries for map display. Those queries will be triggered at the same by GUI manipulate.
(Another points is that issues happened when queries go to **distributed** engine table.)

But I found Clickhouse was shut down by those queried sometimes. Currently, status are:

1. Shut down forever til systemctl restart Clickhouse 
2. Or temporarily showing Error 210.

Then I checked /var/log/clickhouse-server/clickhouse-server.log and /var/log/clickhouse-server/fatal.log.

- I found two different kinds of log showing problem is about memory pointer.
- Should I do some settings to prevent these happened?

![20210107-6](https://user-images.githubusercontent.com/35727603/103844487-f47c2e80-50d4-11eb-8b97-e4b6448398fe.jpg)

![20210107-4](https://user-images.githubusercontent.com/35727603/103844527-03fb7780-50d5-11eb-8181-0ba56b986188.jpg)

**Comment by alexey-milovidov** [comment]
Need to upgrade to the latest ClickHouse release first.

**Comment by tkionshao** [comment]
> Need to upgrade to the latest ClickHouse release first.

But I already use 20.8.7.15, is it not enough?

**Comment by den-crane** [comment]
@tkionshao upgrade to 20.8.11.17

**Comment by alexey-milovidov** [comment]
@tkionshao Did you check the newer version?

**Comment by tkionshao** [comment]
> @tkionshao Did you check the newer version?

Still happened, but moved the image snapshot to other host is working.
I think this is our environment issue, thanks for help.

**Comment by alexey-milovidov** [comment]
Always worth to check `dmesg` for messages about faulty RAM.

**Comment by tkionshao** [comment]
I found the server it crashing using LXC, so I think #20608 is relevant with it.

I tried to use docker contain where Clickhouse is running inside on LXC server. The time of segmentation fault happening is getting lower, but if too many process units are working such as inserting data, dropping data and heavy queries. Segfault is still here. However, when Clickhouse crashing and docker contain stop working, kill then run new container or just start contain again, everything is normal.
